spec/lcm/integration/lcm_end_to_end_spec.rb

```diff
@@ -0,0 +1,255 @@
+require 'pp'
+require 'aws-sdk'
+require 'tempfile'
+require 'csv'
+
+require_relative 'support/constants'
+require_relative 'support/configuration_helper'
+require_relative 'support/lcm_helper'
+require_relative 'shared_examples_for_synchronization_bricks'
+require_relative 'shared_contexts_for_lcm_specs'
+
+# global variables to simplify passing stuff between shared contexts and examples
+$master_projects = []
+$client_projects = []
+$master = false
+
+describe 'the whole life-cycle' do
+  include_context 'lcm bricks'
+
+  describe '1 - Initial Release' do
+    before(:all) do
+      @config_template_path = File.expand_path(
+        '../params/release_brick.json.erb',
+        __FILE__
+      )
+    end
+
+    include_context 'release brick'
+
+    it_behaves_like 'a synchronization brick' do
+      let(:original_project) { @project }
+      let(:projects) { $master_projects }
+      let(:schedules_status) { 'DISABLED' }
+      let(:lcm_managed_tag) { false }
+      let(:update_operations) { [] }
+      let(:update_scripts) { [] }
+      let(:client_id_schedule_parameter) { false }
+      let(:user_group) { true }
+      let(:schedule_diff) do
+        [['+', 'hiddenParams.hidden_msg_from_release_brick', nil],
+         ['+', 'params.msg_from_release_brick', 'Hi, I was set by release brick']]
+      end
+    end
+  end
+
+  describe '2 - Initial Provisioning' do
+    before(:all) do
+      @config_template_path = File.expand_path(
+        '../params/provisioning_brick.json.erb',
+        __FILE__
+      )
+    end
+
+    include_context 'provisioning brick'
+
+    it 'creates client projects only for filtered segments' do
+      expect($client_projects.length).to be 2
+    end
+
+    it_behaves_like 'a synchronization brick' do
+      let(:original_project) { $master }
+      let(:projects) { $client_projects }
+      let(:schedules_status) { 'ENABLED' }
+      let(:lcm_managed_tag) { true }
+      let(:update_operations) { [] }
+      let(:update_scripts) { [] }
+      let(:client_id_schedule_parameter) { true }
+      let(:user_group) { false }
+      let(:schedule_diff) do
+        [['+', 'hiddenParams.hidden_msg_from_provisioning_brick', nil],
+         ['+', 'params.msg_from_provisioning_brick', 'Hi, I was set by provisioning brick']]
+      end
+    end
+  end
+
+  describe '3 - Initial Rollout' do
```

Comment:
Similar code found in lcm_end_to_end_spec.rb:76, lcm_end_to_end_spec.rb:199 (mass = 200)